### PR TITLE
Deploy branch preview and staging env from GH action - Can be reviewed 🔍 

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
Add branch previews and build master into staging env

## Description
This PR adds two github actions:

- Create branch previews when any front-end code is changed (using existing logic, please check)
- Deploy to staging.electricitymaps.com when and branch is merged into master. (perhaps we want to restrict this with some logic too)

TODO: Point branches to electricitymaps-app cloudflare page or leave previews on electricitymaps-staging and remove comment on this PR.

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: .github/workflows/deploy-staging.yml
Comment (VIKTORVAV99) [inline .github/workflows/deploy-staging.yml:40]:
```suggestion
      - name: Build
        working-directory: ./web
        run: pnpm run build
        env:
           VITE_ELECTRICITYMAP_PUBLIC_TOKEN: ${{ secrets.VITE_PUBLIC_ELECTRICITYMAP_PUBLIC_TOKEN }}
```

**Review by VIKTORVAV99 — COMMENTED** [review]
Two small suggested changes.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

**Comment by VIKTORVAV99** [comment]
@tonypls can we merge this or are we waiting for a review from @madsnedergaard as well?

File: .github/workflows/deploy-preview-branch.yml
Comment (madsnedergaard) [inline .github/workflows/deploy-preview-branch.yml:49]: What would the benefit be of switching it back to non-staging project? :)

File: .github/workflows/deploy-staging.yml
Comment (madsnedergaard) [inline .github/workflows/deploy-staging.yml:1]: Will this action actually run on it's on when it has the `on: workflow_call`, but is never called (as far as I could tell)?

**Review by madsnedergaard — COMMENTED** [review]
A few minor questions, otherwise it looks fine to me - and great that you got the secrets working!

**Pull request merged**
